bin/sync/article_updater.rb

```
@@ -0,0 +1,158 @@
+        download_images(attributes[:slug], http_client, working_dir)
+        updated_article = update_canonical_url_on_dev_to(article_id, attributes[:slug])
+        next unless updated_article
+
+        update_article_edited_at(article_id, updated_article)
+      end
+
+    rescue StandardError => e
+      puts "Error processing articles: #{e.message}"
+      raise
+    end
+  end
```

rubocop

[convention] Assignment Branch Condition size for download_new_articles is too high. [<6, 22, 7> 23.85/23]

(Metrics/AbcSize)